csharp/Std.UriTemplate.cs

```diff
@@ -433,6 +450,44 @@ private static bool IsEmpty(SubstitutionType substType, object value)
         }
     }
 
+    private static bool isNativeType(object value)
+    {
+        if (value is string ||
+            value is bool ||
+            value is int ||
+            value is long ||
+            value is float ||
+            value is double ||
+            value is DateTime dt ||
+            value is DateTimeOffset dto)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    private static string convertNativeTypes(object value)
+    {
+        if (value is string ||
+            value is bool ||
+            value is int ||
+            value is long ||
+            value is float ||
+            value is double)
+        {
+            return value.ToString();
+        }
+        else if (value is DateTime dt)
+        {
+            return dt.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ssZ");
+        }
+        else if (value is DateTimeOffset dto)
+        {
+            return dto.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ssZ");
+        }
+        throw new ArgumentException($"Illegal class passed as substitution, found {value.GetType()}");

```

Comment:
```suggestion
        return value switch {
            string str => str,
            bool or int => value.ToString(),
            long or float or double number => number.ToString(CultureInfo.InvariantCulture),
            DateTime dt => dt.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ssZ"),
            DateTimeOffset dto => dto.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ssZ"),
            _ => throw new ArgumentException($"Illegal class passed as substitution, found {value.GetType()}"),
        };
```

we might need to break the number case into multiple cases

Reply:
manually merged

Reply:
We need to use to string overload with the invariant culture for any number type so that it doesn't get localized in some contexts (e.g. q million being represented as 1.000.000 in the us or pi being represented by 3.14 or 3,14 in depending on the locale)

Reply:
uh, ok, let me fix

Reply:
should be fixed now, please have a look

Reply:
with this change, apparently, also the CI warning disappeared!